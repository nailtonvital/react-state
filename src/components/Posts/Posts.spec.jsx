/* eslint-disable testing-library/no-node-access */
import { render, screen } from "@testing-library/react"
import { Posts } from '.'

const props = {
    posts:[
        {
            id: 1,
            title: 't1',
            body: 'b1',
            cover: 'img/c1.png',
        },
        {
            id: 2,
            title: 't2',
            body: 'b2',
            cover: 'img/c2.png',
        },
        {
            id: 3,
            title: 't3',
            body: 'b3',
            cover: 'img/c3.png',
        }
]
    
}

describe('<Posts />', ()=>{
    it('should render posts',()=>{
        render(<Posts {...props}/>)

        expect(screen.getAllByRole('heading', { name: /t/i })).toHaveLength(3)
        expect(screen.getAllByRole('img', { name: /t/i })).toHaveLength(3)
        expect(screen.getAllByText( /b/i )).toHaveLength(3)
        
    })

    it('should not render posts', () => {
        render(<Posts/>)
        expect(screen.queryByRole('heading', { name: /t/i })).not.toBeInTheDocument()
    })

    it('should match snapshot',()=>{
        const { container } = render(<Posts {...props} />)
        expect(container.firstChild).toMatchSnapshot()
    })

})