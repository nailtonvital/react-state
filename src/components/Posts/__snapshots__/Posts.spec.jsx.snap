// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="t1"
      src="img/c1.png"
    />
    <div
      class="post-content"
    >
      <h1>
        t1
      </h1>
      <p>
        b1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="t2"
      src="img/c2.png"
    />
    <div
      class="post-content"
    >
      <h1>
        t2
      </h1>
      <p>
        b2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="t3"
      src="img/c3.png"
    />
    <div
      class="post-content"
    >
      <h1>
        t3
      </h1>
      <p>
        b3
      </p>
    </div>
  </div>
</div>
`;
